lib/diff.js

```diff
@@ -110,16 +110,33 @@ const getAction = ({actual, ideal}) => {
   if (ideal.isRoot && actual.isRoot)
     return null
 
+  // if the versions don't match, it's a change no matter what
+  if (ideal.version !== actual.version)
+    return 'CHANGE'
+
   const binsExist = ideal.binPaths.every((path) => existsSync(path))
 
   // top nodes, links, and git deps won't have integrity, but do have resolved
-  if (!ideal.integrity && !actual.integrity && ideal.resolved === actual.resolved && binsExist)
+  // However! if *neither* has a resolved value, then we need to refetch
+  // to figure out what it even is, so we mark as CHANGE.
```

Comment:
This comment seems out of sync w/ the following code block, which does not mark as change.